contracts/Kernel.sol

```diff
@@ -0,0 +1,136 @@
+pragma solidity ^0.4.21;
+
+import "./ZepToken.sol";
+import "./Release.sol";
+import "./Vouching.sol";
+import "zos-core/contracts/Initializable.sol";
+import "zos-core/contracts/upgradeability/UpgradeabilityProxyFactory.sol";
+import "zeppelin-solidity/contracts/math/SafeMath.sol";
+
+/**
+ * @title Kernel
+ * @dev This contract controls the kernel distributions and versions for ZeppelinOS
+ */
+contract Kernel is Initializable {
+  using SafeMath for uint256;
+
+  // Token used for vouching
+  ZepToken public token;
+
+  // Staking contract
+  Vouching public vouches;
+
+  // Price of registering a new kernel version
+  uint256 public newVersionCost;
+
+  // Fraction of vouches rewarded to the developer of a kernel release
+  uint256 public developerFraction;
+
+  // Keep track of registered kernel releases given their address
+  mapping(address => bool) internal releases;
+
+  /**
+   * @dev Event signaling that a new release has been registered
+   * @param release representing the new Release
+   */
+  event ReleaseRegistered(Release release);
+
+  /**
+  * @dev Guarantees that a given kernel release is registered
+  */
+  modifier onlyWhenRegistered(Release release) {
```

Comment:
just to be sure, did we agree on using `onlyWhen...` or just `when...`?

Reply:
We could change it to just `when`